packages/template-retail-react-app/app/components/_app/index.jsx

```diff
@@ -129,22 +129,30 @@ const App = (props) => {
     // Get the current currency to be used through out the app
     const currency = locale.preferredCurrency || l10n.defaultCurrency
 
-    // Handle creating a new pasket if there isn't one already assigned to the current
+    // Handle creating a new basket if there isn't one already assigned to the current
     // customer.
     const {data: customer} = useCurrentCustomer()
     const {data: baskets} = useCustomerBaskets(
         {parameters: {customerId: customer.customerId}},
         {enabled: !!customer.customerId && !isServer}
     )
     const createBasket = useShopperBasketsMutation('createBasket')
+    const updateBasket = useShopperBasketsMutation('updateBasket')
 
-    // Create a new basket if the current customer doesn't have one.
     useEffect(() => {
+        // Create a new basket if the current customer doesn't have one.
         if (baskets?.total === 0) {
             createBasket.mutate({
                 body: {}
             })
         }
+        // update the basket currency if it doesn't match the current locale currency
+        if (baskets?.baskets?.[0]?.currency && baskets?.baskets?.[0]?.currency !== currency) {
```

Comment:
When `baskets` is undefined, this will evaluate to `true` and it'll mutate an undefined basket.

Reply:
When `baskets` is undefined, `baskets?.baskets?.[0]?.currency` is undefined and the second part of the `&&` is not evaluated. 😉 

```suggestion
        if (baskets?.baskets?.[0]?.currency && baskets.baskets.[0].currency !== currency) {
```
Don't need the null checks twice, though.

Reply:
ahhh, yea, i didn't pay enough attention. ignore my comment!